genesis/genesis.go

```diff
@@ -83,18 +83,24 @@ func NewFile(isQuorum bool, options ...Option) string {
 
 func Save(dataDir string, genesis *core.Genesis, isQuorum bool) error {
 	filePath := filepath.Join(dataDir, FileName)
+	err, jsonStr := ToString(genesis, isQuorum)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(filePath, []byte(jsonStr), 0600)
+}
 
+func ToString(genesis *core.Genesis, isQuorum bool) (error, string) {
 	raw, err := json.Marshal(genesis)
 	if err != nil {
-		return err
+		return err, ""
 	}
 
 	//Quorum hack: add isQuorum field
+	jsonStr := string(raw)
 	if isQuorum {
-		jsonStr := string(raw)
 		idx := strings.Index(jsonStr, ",\"istanbul\"")
 		jsonStr = fmt.Sprintf("%s,\"isQuorum\":true%s", jsonStr[:idx], jsonStr[idx:])
```

Comment:
Try to define new `Genesis` and `ChainConfig` struct ?

Reply:
OK